kafka-1.0.x/src/main/scala/monix/kafka/KafkaConsumerObservable.scala

```diff
@@ -99,6 +106,28 @@ trait KafkaConsumerObservable[K, V, Out] extends Observable[Out] {
     // at most once
     cancelTask.memoize
   }
+
+  /* Returns task that constantly polls the `KafkaConsumer` in case subscriber
+   * is still processing last fed batch.
+   * This allows producer process commit calls and also keeps consumer alive even
+   * with long batch processing.
+   */
+  private def pollConsumer(consumer: KafkaConsumer[K, V]): Task[Unit] = {
+    Task
+      .sleep(config.pollInterval)
+      .flatMap { _ =>
+        if (!isAcked.get()) {
+          Task.evalAsync {
+            consumer.synchronized{
+              blocking(consumer.poll(0))
+            }
+          }
+        } else {
+          Task.unit
+        }
+      }
+      .flatMap(_ => pollConsumer(consumer))
```

Comment:
Is this recursive call necessary? `.loopForever` at line 71 has to do all the work

Reply:
Right, loopForever should be enough here